docs/guides/writingafunction.md

```diff
@@ -1,51 +1,84 @@
 # Writing a Function
 
-TriggerMesh Function CR provides ample opportunities to implement custom events
-flow logic and can act as a source, transformation, or target. Currently,
-Functions supports Python, NodeJS, and Ruby runtimes.
+The TriggerMesh `Function` API provides opportunities to implement custom event flow logic and may act as a source, transformation, or a target. Currently, Python, NodeJS, and Ruby runtimes are supported.
 
-## Prerequisites
+!!! Info "Prerequisites"
+    You need a working TriggerMesh platform installation. See the [installation steps](installation.md).
+    You can verify that the API is available with the following command:
 
-1. K8s cluster and configured kubectl
-1. [Knative Serving and Eventing Operators](https://knative.dev/docs/admin/install/knative-with-operators/)
-1. [TriggerMesh Function](https://github.com/triggermesh/function)
+    ```console
+    $ kubectl get crd |grep function
+    functions.extensions.triggermesh.io        2021-10-06T09:01:33Z
+    ```
 
-## Python function
+!!! Warning
+    The TriggerMesh `Function` API is an opinionated, simple to consume, Function as a Service (FaaS) system. It is aimed to be used for event processing and does not support external dependencies. Functions that may need external dependencies are best served with [Knative Serving](https://knative.dev/docs/getting-started/first-service/).
 
-The Function object spec requires a minimal amount of configuration.
+## Example: A Python function
+
+As an example, let's write a Python function which reads a name from an incoming payload and returns a "Hello" message.
+
+Writing a function requires two steps:
+
+- [ ] Writing a function manifest
+- [ ] Applying the manifest to your Kubernetes cluster
```

Comment:
Here we use `[ ]`, in other places `[x]`.
I have no preference, but we should be consistent.